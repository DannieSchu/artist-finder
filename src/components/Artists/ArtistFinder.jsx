import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist.jsx';
import Header from '../Header/Header.jsx';
import Home from '../Home/Home.jsx';
import { useFindArtist } from '../../hooks/findArtist.jsx';
import withPaging from '../../utils/withPaging.jsx';
import withList from '../../utils/withList.jsx';

const ArtistFinder = ({ page, setTotalPages }) => {
  const { artists, loading, artistSearch, handleChange, handleSubmit } = useFindArtist({ page, setTotalPages });
  const Artists = withList(Artist, 'artists');

  return (
    <>
      <Header 
        onChange={handleChange} 
        onSubmit={handleSubmit} 
        value={artistSearch} 
        placeholder="Find artist" 
        buttonText="Go!"
      />
      {artists.length === 0 && <Home />}
      {artists.length !== 0 && <Artists artists={artists} heading={artistSearch} loading={loading} />}
    </>
  );
};

ArtistFinder.propTypes = {
  page: PropTypes.number.isRequired,
  setTotalPages: PropTypes.func.isRequired
};

export default withPaging(ArtistFinder);
