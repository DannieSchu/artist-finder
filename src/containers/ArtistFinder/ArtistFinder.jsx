import React, { useState } from 'react';
import { fetchArtists } from '../../services/fetchArtists.jsx';
import Artists from '../../components/Artists/Artists.jsx';
import Header from '../../components/Header/Header.jsx';
import Home from '../../components/Home/Home.jsx';

const ArtistFinder = () => {
  const [artistSearch, setArtistSearch] = useState('');
  const [artists, setArtists] = useState([]);
  const [loading, setLoading] = useState(false);
  
  const handleChange = ({ target }) => {
    setArtistSearch(target.value);
  };

  const handleSubmit = event => {
    event.preventDefault();
    setLoading(true);
    fetchArtists(artistSearch)
      .then(res => setArtists(res))
      .then(() => setLoading(false));
  };

  return (
    <>
      <Header onChange={handleChange} 
        onSubmit={handleSubmit} 
        value={artistSearch} 
        placeholder="Find artist" 
        buttonText="Go!"
      />
      {artists.length === 0 && <Home />}
      {artists.length !== 0 && <Artists artists={artists} loading={loading} />}
    </>
  );
};

export default ArtistFinder;
