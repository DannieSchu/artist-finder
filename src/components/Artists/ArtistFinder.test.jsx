import React from 'react';
import { shallow } from 'enzyme';
import ArtistFinder from './ArtistFinder.jsx';

describe('ArtistFinder', () => {
  it('matches a snapshot', () => {
    const artists = [{
      id: 'gh8932hg3gee',
      name: 'Leonard Cohen',
      origin: 'Canada',
      birth: '1934-09-21',
      death: '1934-09-21',
    },
    {
      id: '6943w0utw0ghjvdr',
      name: 'Leonard Cohen',
      origin: 'Canada',
      birth: '1934-09-21',
      death: '1934-09-21' 
    },
    {
      id: 't430tu340grjid',
      name: 'Leonard Cohen',
      origin: 'Canada',
      birth: '1934-09-21',
      death: '1934-09-21' 
    }];
    
    const wrapper = shallow(
      <ArtistFinder artists={artists} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
