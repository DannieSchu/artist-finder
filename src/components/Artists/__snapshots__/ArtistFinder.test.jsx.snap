// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistFinder matches a snapshot 1`] = `
<Fragment>
  <ArtistFinder
    artists={
      Array [
        Object {
          "birth": "1934-09-21",
          "death": "1934-09-21",
          "id": "gh8932hg3gee",
          "name": "Leonard Cohen",
          "origin": "Canada",
        },
        Object {
          "birth": "1934-09-21",
          "death": "1934-09-21",
          "id": "6943w0utw0ghjvdr",
          "name": "Leonard Cohen",
          "origin": "Canada",
        },
        Object {
          "birth": "1934-09-21",
          "death": "1934-09-21",
          "id": "t430tu340grjid",
          "name": "Leonard Cohen",
          "origin": "Canada",
        },
      ]
    }
    page={1}
    setTotalPages={[Function]}
  />
  <Paging
    incrementPage={[Function]}
    page={1}
    totalPages={1}
  />
</Fragment>
`;
